wtpython/__main__.py

```diff
@@ -5,24 +5,49 @@
 
 import pyperclip
 from rich import print
+from rich.markdown import HorizontalRule
+from rich.traceback import Traceback
 
 from wtpython import SearchError
 from wtpython.backends.stackoverflow import StackOverflowFinder
 from wtpython.display import Display, store_results_in_module
 from wtpython.settings import GH_ISSUES, MAX_SO_RESULTS
 
 
+def trim_exception_traceback(tb: traceback) -> traceback:
+    """
+    Trim the traceback to remove extra frames
+
+    Because of the way we are currently running the code, any traceback
+    created during the execution of the application will be include the
+    stack frames of this application. This function removes all the stack
+    frames from the beginning of the traceback until we stop seeing `runpy`.
+    """
+    seen_runpy = False
+    while tb is not None:
+        cur = tb.tb_frame
+        filename = cur.f_code.co_filename
+        if "runpy" in filename:
+            seen_runpy = True
+        elif seen_runpy and "runpy" not in filename:
+            break
+        tb = tb.tb_next
+
+    return tb
+
```

Comment:
How about

```python
stack_summary = traceback.extract_tb(tb)
end_of_runpy = max([
    i for i, stack in enumerate(stack_summary, 1)
    if 'runpy' in stack.filename
])
print(''.join(traceback.format_list(stack_summary[end_of_runpy:])))
```

Reply:
obviously, not "print" but we'd return the tb as needed

Reply:
`traceback.extract_tb(tb)` doesn't give us the frame objects, but rather just a summary of the frames. We can't actually build up a traceback object from this AFAIK.